design/FY2020/NFP-ZoneAirTransferModel.md

```diff
@@ -0,0 +1,143 @@
+
+NA
+
+## Overview ##
+
+This work will implement code changes that allow alternative approach of the zone air mass flow balance calculation method that does not alter the simple flow objects mass flow. 
+The fundamental zone mass balance equation remains the same but what this feature requesting is which zone air mass flow component needs to be adjusted to guarantee zone air mass flow conservation.
+
+## Approach ##
+
+Current zone air mass flow conservation equation is formulated to solve the following equation by adjusting the Zone Mixing flow, and zone infiltration flow sequentially.
+
+ 0.0 = [m_{sup} - m_{exh} - m_{ret} + m_{zmreceiving} - m_{zmsource} + m_{inf}]
+	
+where,
+m_{sup} = Zone Supply Air Mass Flow Rate, [kg/s]
+m_{exh} = Zone Exhaust Air Mass Flow Rate, [kg/s]
+m_{ret} = Zone Return Air Mass Flow Rate, [kg/s]
+m_{zmreceiving} = Zone Mixing Receiving Air Mass Flow Rate, [kg/s]
+m_{zmsource} = Zone Mixing Source Air Mass Flow Rate, [kg/s]
+m_{inf} = Zone Infiltration Air Mass Flow Rate, [kg/s]
+
+The existing procedure first attempts to solve the above equation by adjusting the Zone Mixing flow only and if that does not satisfy the above equation, then the user specified zone infiltration air flow rate is adjusted as needed.  The zone infiltration air flow can be adjusted using two different methods depending user preference. The zone infiltration air flow adjusting method is a choice input field in the ZoneAirMassFlowConservation object.
+
+The requested feature can be implemented by adjusting the return air mass flow but the return air mass flow rate can be varied between the minimum value of zero and a maximum value of the zone return air maximum flow rate. Since varying the return air mass flow only provides limiting flexibility in achieving balanced flow for the range of practical applications, the zone air flow mass balance requires two different solution scenarios: one when the zone is under positive pressure and when the zone is under negative pressure.
+
+Solving the zone air flow mass balance equation for zone return air mass flow:
+
+m_{ret} = [m_{sup} - m_{exh} + m_{zmreceiving} - m_{zmsource} + m_{inf}]
+ 
+Depending on the sign of the value of return air mass flow rate calculated using the above equation two different cases that require different solution scheme are formulated:
+
+Case I:  [m_{ret} >= 0.0]
+
+         The zone is under positive pressure and requires return air mass flow rate to balance the zone air flow. The return air flow rate will be adjusted within the bounds of 0 and maximum return air flow rate to balance the zone air mass flow.
+
+Case II: [m_{ret} < 0.0]
+            
+         The zone is under negative pressure and requires additional supply air flow to balance the zone air flow. In this case the return air mass flow rate will be reset to zero and the supply air flow rate is increased proportionally.
+		 [m_{ret} = 0.0;]
+		 [m_{sup} += m_{ret};]
+		 
```

Comment:
How are constant volume HVAC systems going to respond to these 2 cases? I assume OA will make up the difference between supply and return flow rates. What happens when there is no OA system? In that case the supply and return flows should match.

Reply:
I am assuming that the two cases should work for constant volume and VAV systems and have to confirmed through testing.

Reply:
Regarding "What happens when there is no OA system?", I have not thought enough if it needs a special treatment. One issue is that supply and return flows may not match for zones that has exhaust fan flow, ZoneMixing and Infiltration flows. You have to adjust either the supply or the return flows as needed.

Reply:
1. For systems with no outdoor air, the existing return flow calculations force the return to match the corresponding supply flow for a given zone. This balance must be maintained.
2. For constant volume systems, are you suggesting that the system flow could be increased by this? That's going to run up against the terminal unit max flow rate.
3. For VAV systems, increasing the supply flow could cause temperature control problems.

Reply:
I suspected I may run into control problem when the supply air flow rates is altered. In some cases return air can be zero and we may not have other variable to change unless infiltration air flow rate is adjusted instead of supply air. In fact my original NFP draft was to adjust return air and infiltration to balance the flow while keeping the ZoneMixing object flow as specified and letting the control system deal with supply air flow rate.

I would rather adjust return and infiltration air flow rates only to balance the system and let the control system determine the supply air flow rate. What do you think? @mjwitte

Reply:
That seems safer, but that question should be answered by those who are requesting this feature and the specific use cases that it will be applied to.

Reply:
OK, I will seek input from those requested the feature.

Reply:
Regarding Case II here. 
For VAV terminals, it does seem possible to consider exhaust fan flows when determining damper position, but that is beyond scope I think.   

I think when return air wants to go negative there is a sizing or input consistency problem.  The user should have set the ventilation requirement in DesignSpecification:OutdoorAir object high enough to supply make up air for exhaust fan and mixed air transfers.   For Case II, I would say don't do anything beyond setting return air to zero.  The existing unbalanced flow warning should catch this okay.